gnomad_hail/utils/variant_qc.py

```diff
@@ -262,80 +275,481 @@ def quantiles_to_bin_boundaries(quantiles: List[int]) -> Dict:
+    truth_data = truth_ht[ht.key].truth_data
+    fam = fam_stats_ht[ht.key]
+
+    return dict(
+        n_clinvar=hl.agg.count_where(hl.is_defined(clinvar)),
+        n_de_novos_hq=hl.agg.sum(fam.n_de_novos_hq),
+        n_de_novos_adj=hl.agg.sum(fam.n_de_novos_adj),
+        n_de_novo=hl.agg.sum(fam.n_de_novos_raw),
+        n_trans_singletons=hl.agg.filter(ht.ac_raw == 2, hl.agg.sum(fam.n_transmitted_raw)),
+        n_untrans_singletons=hl.agg.filter((ht.ac_raw < 3) & (fam.unrelated_qc_callstats.AC[1] == 1),
+                                           hl.agg.sum(fam.tdt.u)),
+        ## n_train_trans_singletons=hl.agg.filter((ht.ac_raw == 2) & rank_ht.positive_train_site, hl.agg.sum(fam.n_transmitted_raw)),
+        n_omni=hl.agg.count_where(truth_data.omni),
+        n_mills=hl.agg.count_where(truth_data.mills),
+        n_hapmap=hl.agg.count_where(truth_data.hapmap),
+        n_kgp_phase1_hc=hl.agg.count_where(truth_data.kgp_phase1_hc)
+    )
+
+
+def compute_aggregate_binned_data(
+        bin_ht: hl.Table,
+        agg_func: Callable = default_score_bin_agg,
+        checkpoint_path: Optional[str] = None,
+        **kwargs
+) -> hl.Table:
+    """
+    Aggregates a Table that has been annotated with bins based on quantiles (`compute_quantile_bin` or
+    `create_binned_ht`). The table will be grouped by bin_id (bin, biallelic, etc.), contig, snv, bi_allelic and
+    singleton. Then for each grouping, min/max of `score` will be computed and any other desired aggregations.
+
+    Automatic aggregations that will be done are:
+        `min_score` - minimun of score annotation per group
+        `max_score` - maiximum of score annotation per group
+        `n` - count of variants per group
+        `n_ins` - count of insertion per group
+        `n_ins` - count of insertion per group
+        `n_del` - count of deletions per group
+        `n_ti` - count of transitions per group
+        `n_tv` - count of trnasversions per group
+        `n_1bp_indel` - count of one base pair indels per group
+        `n_mod3bp_indel` - count of indels with a length divisible by three per group
+        `n_singleton` - count of singletons per group
+        `fail_hard_filters` - count of variants per group with QD < 2 | FS > 60 | MQ < 30
+        `n_vqsr_pos_train` - count of variants that were a VQSR positive train site per group
+        `n_vqsr_neg_train` - count of variants that were a VQSR negative train site per group
+
+    Requires that `bin_ht` be annotated with an `info` struct that includes QD, FS, and MQ in order to add an
+    annotation for `fail_hard_filters`
+
+    :param Table bin_ht: Input Table with a `bin_id` annotation
+    :param callable agg_func: Function that returns a dict of any additional aggregations to perform
+    :param str checkpoint_path: If provided an intermediate checkpoint table is created with all required annotations before shuffling.
+    :return Table grouped by rank(s) and with counts of QC metrics
```

Comment:
`:return` needs to have a colon after it (`:return:`) to be interpreted correctly.